SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Build/Build.cs

```diff
@@ -86,10 +82,55 @@ public static void Main(string[] args)
             var noCompile = args.Count(arg => arg.ToLowerInvariant() == "-nocompile") > 0;
             var additionalUATArgs = string.Join(" ", args.Skip(4).Where(arg => arg.ToLowerInvariant() != "-nocompile"));
 
-            var stagingDir = Path.GetFullPath(Path.Combine("../spatial", "build", "unreal"));
-            var outputDir = Path.GetFullPath(Path.Combine("../spatial", "build", "assembly", "worker"));
+            var stagingDir = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(projectFile), "../spatial", "build", "unreal"));
+            var outputDir = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(projectFile), "../spatial", "build", "assembly", "worker"));
             var baseGameName = Path.GetFileNameWithoutExtension(projectFile);
 
+            // Get UNREAL_HOME without environment variable.
+            Console.WriteLine("Finding Unreal Engine build.");
+            string uproject = File.ReadAllText(projectFile, Encoding.UTF8);
+            string engineAssociationRegex = "..{8}-.{4}-.{4}-.{4}-.{12}.";
+            Match engineAssociationMatch = Regex.Match(uproject, engineAssociationRegex);
+
+            string unrealEngine = "";
+
+            // If we found an engine association within the .uproject, query the registry for the engine directory.
+            if (engineAssociationMatch.Success)
+            {
+                string engineAssociation = engineAssociationMatch.Value;
+                Console.WriteLine("Engine Association: " + engineAssociation);
+
+                string unrealEngineBuildKey = "HKEY_CURRENT_USER\\Software\\Epic Games\\Unreal Engine\\Builds";
+                unrealEngine = Registry.GetValue(unrealEngineBuildKey, engineAssociation, "").ToString();
+            }
+            else // If we couldn't find the Engine Association. Climb the parent directories looking for the "Engine".
+            {
+                DirectoryInfo currentDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+                while (currentDir.Parent != null && unrealEngine == "")
```

Comment:
Use `string.IsNullOrEmpty` instead of `== ""` (Same below for the next `if` statement)

Reply:
Nice, cheers.